clients/src/main/java/org/apache/kafka/common/protocol/ProtoUtils.java

```diff
@@ -27,7 +27,7 @@ private static Schema schemaFor(Schema[][] schemas, int apiKey, int version) {
         if (apiKey < 0 || apiKey > schemas.length)
             throw new IllegalArgumentException("Invalid api key: " + apiKey);
         Schema[] versions = schemas[apiKey];
-        if (version < 0 || version > versions.length)
+        if (version < 0 || version >= versions.length)
```

Comment:
Instead of using length of the versions list, would it make sense to check if the value is greater than the latest version?